dask/bag/core.py

```diff
@@ -2498,3 +2478,121 @@ def to_dataframe(seq, columns, dtypes):
         seq = list(seq)
     res = pd.DataFrame(seq, columns=list(columns))
     return res.astype(dtypes, copy=False)
+
+
+def repartition_npartitions(bag, npartitions):
+    """ Changes the number of partitions of the bag.
+
+    This can be used to reduce or increase the number of partitions
+    of the bag.
+    """
+    new_name = "repartition-%d-%s" % (npartitions, tokenize(bag, npartitions))
+    if npartitions == bag.npartitions:
+        return bag
+    elif bag.npartitions > npartitions:
+        ratio = bag.npartitions / npartitions
+        new_partitions_boundaries = [
+            int(old_partition_index * ratio)
+            for old_partition_index in range(npartitions + 1)
+        ]
+        return _repartition_from_boundaries(bag, new_partitions_boundaries, new_name)
+    else:  # npartitions > bag.npartitions
+        div, mod = divmod(npartitions, bag.npartitions)
+        nsplits = [div] * bag.npartitions
+        nsplits[-1] += mod
+        return _split_partitions(bag, nsplits, new_name)
+
+
+def repartition_size(bag, size):
+    """
+    Repartition bag so that new partitions have approximately `size` memory usage each
+    """
+    if isinstance(size, str):
+        size = parse_bytes(size)
+    size = int(size)
+    mem_usages = bag.map_partitions(total_mem_usage).compute()
+
+    # 1. split each partition that is larger than partition size
+    nsplits = [1 + mem_usage // size for mem_usage in mem_usages]
+    if any((nsplit > 1 for nsplit in nsplits)):
+        split_name = "repartition-split-{}-{}".format(size, tokenize(bag))
```

Comment:
I recommend that we exclude the full integer size here and instead put that in the tokenization call.  These names are used in things like diagnostics and the dashboard.  Having values like `repartition-split-10248258572` show up in a progress bar is probably not ideal.  If you want to include the text version I'm ok with that, but I would probably defer to just tokenizing everything.